### PR TITLE
`macros`: match unused vars if logging is disabled

### LINKED ISSUE #494: semver-breaking: "Unused variable" errors with deny(unused).
The following used to build fine with defmt 0.2.1, and now fails with defmt 0.2.2. This is due to #477 

```rust
#![deny(unused)]
let x = 1; // error: unused variable: `x` if no "defmt-xxx" features are enabled
info!("x={}", x);
```

### PR DESCRIPTION
**Pull request by Urhengulas**
Fixes #494

### CODE REVIEW

**Comment by Urhengulas** [comment]
I am a bit unsure how to test this. Running the snapshot tests with `--no-default-features` would probably work, since it then would emit the unused-vars-warnings which should mismatch the snapshots, but this seems a bit over the top.

File: macros/src/lib.rs [hunk 1]
```diff
@@ -512,6 +526,7 @@ fn log(level: Level, log: FormatArgs) -> TokenStream2 {
 
     let sym = mksym(&ls, level.as_str(), true);
     let logging_enabled = cfg_if_logging_enabled(level);
+    let logging_disabled = cfg_if_logging_disabled(level);
     quote!({
         #[#logging_enabled] {
```
Comment (jonas-schievink) [inline macros/src/lib.rs:531]: If you change this to

```suggestion
        #[cfg(#logging_enabled)] {
```

and change the code above to not include `cfg()`, you don't need 2 separate functions and `quote!` invocations, that should be a bit easier to read

File: macros/src/lib.rs [hunk 2]
```diff
@@ -525,6 +540,11 @@ fn log(level: Level, log: FormatArgs) -> TokenStream2 {
                 }
             }
         }
+        // if logging is disabled match args, so they are not unused
+        #[#logging_disabled]
```
Comment (jonas-schievink) [inline macros/src/lib.rs:544]:
```suggestion
        #[cfg(not(#logging_enabled))]
```

File: macros/src/lib.rs (hunk 1 above)
Comment (Urhengulas) [reply macros/src/lib.rs:531]: That's smart 🥲

**Pull request merged**